controller/search.go

```diff
@@ -443,3 +449,31 @@ func (c *SearchController) Codebases(ctx *app.CodebasesSearchContext) error {
 	setPagingLinks(response.Links, buildAbsoluteURL(ctx.Request), len(matchingCodebases), offset, limit, totalCount, "url="+ctx.URL)
 	return ctx.OK(&response)
 }
+
+func convertGithubURL(urlRaw string) (string, error) {
+	// if the URL is https then we don't need to make any changes
+	if strings.HasPrefix(urlRaw, "https") {
+		return urlRaw, nil
+	}
+
+	validGit := regexp.MustCompile(`^(https|git)(:\/\/|@)([^\/:]+)[\/:]([^\/:]+)\/(.+).git$`)
+	if !validGit.MatchString(urlRaw) {
+		return "", fmt.Errorf("invalid URL: %v", urlRaw)
+	}
+	components := validGit.FindStringSubmatch(urlRaw)
+	l := len(components)
+
+	repository := components[l-1] + ".git"
```

Comment:
@surajssd can't include `.git` as part of last component along with repository ?